### PR TITLE
Improve error handling with network connections (Boost ASIO)

### PR DESCRIPTION
**Pull request by dnsmichi**
refs #7041

### CODE REVIEW

File: lib/remote/apilistener.cpp [hunk 1]
```diff
@@ -605,11 +616,11 @@ void ApiListener::NewClientHandlerInternal(boost::asio::yield_context yc, const
 			if (client->async_fill(yc[ec]) == 0u) {
 				if (identity.IsEmpty()) {
 					Log(LogInformation, "ApiListener")
-						<< "No data received on new API connection. "
+						<< "No data received on new API connection " << conninfo << ". "
```
Comment (Al2Klimov) [inline lib/remote/apilistener.cpp]:
```
[2019-06-04 08:35:10 -0600] information/ApiListener: No data received on new API connection from [10.211.55.2]:63755. Ensure that the remote endpoints are properly configured in a cluster setup.
```

File: lib/remote/apilistener.cpp [hunk 2]
```diff
@@ -537,6 +530,24 @@ void ApiListener::NewClientHandlerInternal(boost::asio::yield_context yc, const
...
+	boost::system::error_code ec;
+
+	sslConn.async_handshake(role == RoleClient ? sslConn.client : sslConn.server, yc[ec]);
+
+	if (ec) {
+		// https://github.com/boostorg/beast/issues/915
+		// Google Chrome 73+ seems not close the connection properly, https://stackoverflow.com/questions/56272906/how-to-fix-certificate-unknown-error-from-chrome-v73
+		if (ec == asio::ssl::error::stream_truncated) {
+			Log(LogNotice, "ApiListener")
+				<< "TLS stream was truncated, ignoring connection from " << conninfo;
```
Comment (Al2Klimov) [inline lib/remote/apilistener.cpp]:
```
[2019-06-04 08:35:56 -0600] notice/ApiListener: TLS stream was truncated, ignoring connection from from [10.211.55.2]:63763
```

File: lib/remote/apilistener.cpp [hunk 3]
```diff
@@ -537,6 +530,24 @@ void ApiListener::NewClientHandlerInternal(boost::asio::yield_context yc, const
...
+			return;
+		}
+
+		Log(LogCritical, "ApiListener")
+			<< "Client TLS handshake failed (" << conninfo << "): " <<  ec.message();
```
Comment (Al2Klimov) [inline lib/remote/apilistener.cpp]:
```
[2019-06-04 08:38:09 -0600] critical/ApiListener: Client TLS handshake failed (from [10.211.55.2]:63769): wrong version number
```

**Comment by dnsmichi** [comment]
Applied, thanks. Wasn't sure about the changes, so a second pair of eyes helps here :)

**Pull request merged**